packages/core-flows/src/region/steps/update-regions.ts

```diff
@@ -0,0 +1,54 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  FilterableRegionProps,
+  IRegionModuleService,
+  UpdatableRegionFields,
+} from "@medusajs/types"
+import { getSelectsAndRelationsFromObjectArray } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type UpdateRegionsStepInput = {
+  selector: FilterableRegionProps
+  update: UpdatableRegionFields
+}
+
+export const updateRegionsStepId = "update-region"
+export const updateRegionsStep = createStep(
+  updateRegionsStepId,
+  async (data: UpdateRegionsStepInput, { container }) => {
+    const service = container.resolve<IRegionModuleService>(
+      ModuleRegistrationName.REGION
+    )
+
+    const { selects, relations } = getSelectsAndRelationsFromObjectArray([
+      data.update,
+    ])
+
+    const prevData = await service.list(data.selector, {
```

Comment:
Are this and the operation below atomic? if not, we can easily end up with a race condition where you list the previous data, another workflow updates the same region, then this update fails, and you revert to the oldest data that is no longer correct.

I have quite a few questions around race conditions and resilience of workflows that maybe I'll talk to Carlos about :)

Reply:
we have an on going discussion around loack, you can discuss it with carlos too